integration_test/Dialect/Handshake/basic.cpp

```diff
@@ -0,0 +1,61 @@
+//===- basic.cpp - Verilator software driver ------------------------------===//
+//
+// Part of the LLVM Project, under the Apache License v2.0 with LLVM Exceptions.
+// See https://llvm.org/LICENSE.txt for license information.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+//
+//===----------------------------------------------------------------------===//
+
+#include "Vtop.h"
+
+int main(int argc, char **argv) {
```

Comment:
Why does this need its own verilator driver?  I thought the rtl-sim script had one?

Reply:
This is what I was saying during that simulator discussion we were having. This driver is necessary to provide stimulus to the design.

Reply:
It's it possible to declarativize this like we do for the other testing tools?  I don't see how this is different.

Reply:
See how the stimulus (and checking) is different from "real" simulators? From a 10k foot view, it's only different in the language -- C++ vs SystemVerilog.

Making a stimulus and checking language which compiles to either C++ or SystemVerilog is something I also brought up in that discussion. I think there's a big opportunity to create a 'verification' dialect to do exactly that! But it'd be a big project without any input language. Starting with a simple dialect to fill our modest needs is probably a good start.

Either way, I think this is a quite a large ask.

Reply:
A verification dialect sounds really interesting, but out of the scope for what I'm trying to achieve here.

Is this the correct way to write a Verilator test case? I've written testbenches in Verilog that instantiate the device under test, provide some stimuli, and print/assert on the DUT outputs. My understanding of Verilator is we are doing the same thing here, but with the driver in C++.

Reply:
Frankly, I have no idea. I’m not a verification expert so I only write basic, ad-hoc SV tests. I’ve mostly used Verilator to connect up to larger pieces of software, not driven signals individually.